core/arch/emit_utils_shared.c

```diff
@@ -4830,16 +4861,28 @@ emit_do_syscall_common(dcontext_t *dcontext, generated_code_t *code, byte *pc,
     APP(&ilist, instr_create_save_to_tls(dcontext, DR_REG_R0, TLS_REG0_SLOT));
     /* XXX: should have a proper patch list entry */
     *syscall_offs += THUMB_LONG_INSTR_SIZE;
-#elif defined(AARCH64)
-    /* For AArch64, we need to save both x0 and x1 into SLOT 0 and SLOT 1
+
+    /* For AArch64/RISCV64, we need to save both x0/a0 and x1/a1 into SLOT 0 and SLOT 1
```

Comment:
Again: it'd be clearer to mention both archs separately.

Reply:
Since there is an `#if defined(ARM)` above at the same level, I separated the comment into two, which seems cleaner.